src/UI/Components/Diagram/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import { ServiceInstanceModel, ServiceModel } from "@/Core";
+
+export const extractRelationsIds = (
+  service: ServiceModel,
+  instance: ServiceInstanceModel
+): string[] => {
+  const relationKeys = service.inter_service_relations?.map(
+    (relation) => relation.name
+  );
+  if (!relationKeys) {
+    return [];
+  }
+  if (instance.candidate_attributes !== null) {
```

Comment:
I'd have the "state" as an argument for the toggling, I think it might be easier.

something like this :
`instance[attribute_state]`

Reply:
I thought that would require additional cast of type on it, but just checked and that would work. I will change it in then on current branch that I'm working on if that's okay with you